packages/shared/src/hooks/usePostMenuActions.ts

```diff
@@ -113,23 +114,28 @@ export const usePostMenuActions = ({
     onConfirmDeletePost: canDelete ? deletePostPrompt : null,
     onPinPost: canPin ? onPinPost : null,
     onToggleDownvotePost: () => {
+      if (!user) {
+        showLogin(AuthTriggers.Downvote);
+        return;
+      }
+
       if (post.downvoted) {
         trackEvent(
           postAnalyticsEvent(AnalyticsEvent.RemovePostDownvote, post, {
             extra: { origin },
           }),
         );
 
-        return cancelPostDownvote({ id: post.id });
-      }
-
-      trackEvent(
-        postAnalyticsEvent(AnalyticsEvent.DownvotePost, post, {
-          extra: { origin },
-        }),
-      );
+        cancelPostDownvote({ id: post.id });
```

Comment:
Why did we change this part?

Reply:
Looks like a refactor to use if else instead of early return. It should be fine either way but additional changes in PR often make reviews harder because you gotta wrap your hand around what is going on instead of just looking for additions and actual fix.

Reply:
I preferred to not change so much as i agree makes it look like more has changed. But, I had to change the markup as i was getting tsc compile errors with different types of returns basically

Reply:
Oh got it, I am fine with it 👍 

Only issue is that per `UsePostMenuActions` this function should be `async` and return promise so to keep it in line we should await mutations inside if else.

Reply:
Another example where our tsc does not catch that type inconsistency. 😢